### PR TITLE
soapdenovo-trans: Create new package.

### PR DESCRIPTION
**Pull request by lingnanyuan**
Add new package: soapdenovo-trans.

### CODE REVIEW

File: var/spack/repos/builtin/packages/soapdenovo-trans/package.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    version('1.0.4', 'a3b00b0f743b96141c4d5f1b49f2918c')
+
+    def install(self, spec, prefix):
+        makefile = './src/Makefile'
+        filter_file(r'CFLAGS=         -O3 -fomit-frame-pointer -static',
+                            'CFLAGS=         -O3 -fomit-frame-pointer',
+                            makefile)
+
+        sh = which('sh')
+        sh('./make.sh')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/soapdenovo-trans/package.py]: It looks like `make.sh` just runs:
```bash
cd src/
make
make 127mer=1
```
I would rather this be written as a `MakefilePackage` where we set:
```python
build_directory = 'src'
```
We can edit the Makefile in the `edit` phase and run `make` twice in the `build` phase. Let me know if you have any questions. I can help with this if you need it.

Comment (lingnanyuan) [reply var/spack/repos/builtin/packages/soapdenovo-trans/package.py]: I tried this, but it seems that '127mer=1' is not a make target, so when I run the second make('127mer=1') in build phase, spack doesn't recognize it and report some error like this:

==> Error: ProcessError: Command exited with status 2:
    'make' '-j24' '127mer=1'

47 errors found in build log:
     42    scaffold.c:73:4: warning: implicit declaration of function ‘RPKMStat’ [-Wimplicit-function-declaration]
     43        RPKMStat(graphfile);
     44        ^~~~~~~~
../SOAPdenovo-Trans-31mer compilation done.
     46    ==> 'make' '-j24' '127mer=1'
     47    ../SOAPdenovo-Trans-127mer cleaning done.
  >> 48    gcc: error: arc.o: No such file or directory
  >> 49    gcc: error: attachPEinfo.o: No such file or directory
  >> 50    gcc: error: bubble.o: No such file or directory

Have no idea what's going on.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/soapdenovo-trans/package.py]: What does your build phase look like?

Comment (lingnanyuan) [reply var/spack/repos/builtin/packages/soapdenovo-trans/package.py]: never mind, I set parallel=False and run again, it works now.

**Pull request merged**